libdnf/rpm/package.cpp

```diff
@@ -0,0 +1,402 @@
+/*
+Copyright (C) 2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+
+#include "libdnf/rpm/package.hpp"
+#include "reldep_list_impl.hpp"
+#include "sack_impl.hpp"
+#include "solv/package-private.hpp"
+
+
+
+inline static std::string
+cstring2string(const char * input)
+{
+    return input ? std::string(input) : std::string();
+}
+
+
+namespace libdnf::rpm {
+
+const char *
+Package::get_name_cstring() const noexcept
+{
+    Pool * pool = sack->pImpl->pool;
+    return libdnf::rpm::solv::get_name(pool, id);
```

Comment:
The code is in the `libdnf::rpm` namespace. Please use `solv::get_name(pool, id)`. The same is repeated in next methods.

Second possibility is using absolute namespace path. Starts with `::`. Example: `::libdnf::rpm::solv::get_name(pool, id)`. I do not recommend absolute path in these cases.

Using fullpath without `::` is possible but it is considered as relative path. If the relative path does not exist in current namespace, then compiler looks for the same relative path in one level up. And it can repeat up to the global namespace. So, it is working. But I think, the shortest method `solv::get_name(pool, id)` is what we want in this case.

Reply:
Replaced